### PR TITLE
14.0 mig server environment

### PR DESCRIPTION
**Pull request by mourad-ehm**

### CODE REVIEW

File: server_environment/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+companion field ``<field>_env_default`` is created. This field is not
+environment-dependent. It's a fallback value used when no key is set in
+configuration files / environment variable.
+
+When the default field is used, the field is made editable on Odoo.
+
+Note: empty environment keys always take precedence over default fields
+
+
+Keychain integration
```
Comment (bealdav) [inline server_environment/readme/CONFIGURE.rst]: It seems keychain no more exists replaced by ...

File: server_environment/server_env.py [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (bealdav) [inline server_environment/server_env.py]: probably could be made shorter

File: server_environment/system_info.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (bealdav) [inline server_environment/system_info.py:17]: probably could be made shorter header

File: server_environment/server_env.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+    config = Serialized()
+
+    @classmethod
+    def _build_model(cls, pool, cr):
+        """Add columns to model dynamically
+        and init some properties
+
+        """
+        ModelClass = super(ServerConfiguration, cls)._build_model(pool, cr)
```
Comment (bealdav) [inline server_environment/server_env.py]:
```suggestion
        ModelClass = super()._build_model(pool, cr)
```

**Review by bealdav — CHANGES_REQUESTED** [review]
Thanks for your contribution. Some typo remarks only

**Review by bealdav — APPROVED** [review]

**Review by sbejaoui — APPROVED** [review]
LGTM, thank you

**Pull request merged**